File: iOS/XtreamPlayer/Domain/Entities/Stream.swift
// iOS/XtreamPlayer/Domain/Entities/Stream.swift
import Foundation

enum StreamType: String, Codable {
    case live
    case vod
    case series
}

struct Stream: Codable, Identifiable, Equatable, Hashable {
    let id: Int
    let num: Int
    let name: String
    let streamIcon: String
    let streamId: String
    let categoryId: String
    let categoryName: String
    let directSource: String
    let rating: Double?
    let added: Date
    var type: StreamType

    init(
        id: Int,
        num: Int,
        name: String,
        streamIcon: String,
        streamId: String,
        categoryId: String,
        categoryName: String,
        directSource: String,
        rating: Double? = nil,
        added: Date = Date(),
        type: StreamType = .live
    ) {
        self.id = id
        self.num = num
        self.name = name
        self.streamIcon = streamIcon
        self.streamId = streamId
        self.categoryId = categoryId
        self.categoryName = categoryName
        self.directSource = directSource
        self.rating = rating
        self.added = added
        self.type = type
    }
}

File: Android/app/src/main/kotlin/xtream/emin/player/domain/entities/Stream.kt
// Android/app/src/main/kotlin/com/xtream/player/domain/entities/Stream.kt
package xtream.emin.player.domain.entities

/**
 * The kind of content a [Stream] represents. Xtream Codes exposes three
 * distinct catalog types via player_api.php; we normalize them into one
 * entity shape with a discriminator so the rest of the app (favorites,
 * history, player) can treat them uniformly.
 */
enum class StreamType {
    LIVE,
    VOD,
    SERIES
}

/**
 * A single playable (or browsable, for series) item from the Xtream Codes
 * catalog - a live channel, a VOD movie, or a series entry.
 */
data class Stream(
    val streamId: String,
    val name: String,
    val streamType: StreamType,
    val categoryId: String? = null,
    val streamIcon: String? = null,
    val containerExtension: String? = null,
    val added: String? = null,
    val rating: String? = null,
    val isAdult: Boolean = false
)
